exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/StatisticsMergeBatch.java

```diff
@@ -0,0 +1,532 @@
+    }
+  }
+
+  private void createKeyColumn(String name, LogicalExpression expr, Map<MaterializedField, ValueVector> parentMap)
+      throws SchemaChangeException {
+    ErrorCollector collector = new ErrorCollectorImpl();
+
+    LogicalExpression mle = ExpressionTreeMaterializer.materialize(expr, incoming, collector,
+        context.getFunctionRegistry());
+
+    MaterializedField outputField = MaterializedField.create(name, mle.getMajorType());
+    ValueVector vector = TypeHelper.getNewVector(outputField, oContext.getAllocator());
+    container.add(vector);
+
+    if (collector.hasErrors()) {
+      throw new SchemaChangeException("Failure while materializing expression. "
+          + collector.toErrorString());
+    }
+    parentMap.put(outputField, vector);
+  }
+
+  private ValueVector addMapVector(String name, MapVector parent, LogicalExpression expr) throws SchemaChangeException {
+    ErrorCollector collector = new ErrorCollectorImpl();
+
+    LogicalExpression mle = ExpressionTreeMaterializer.materialize(expr, incoming, collector,
+        context.getFunctionRegistry());
+
+    Class<? extends ValueVector> vvc =
+        TypeHelper.getValueVectorClass(mle.getMajorType().getMinorType(), mle.getMajorType().getMode());
+    ValueVector vector = parent.addOrGet(name, mle.getMajorType(), vvc);
+
+    if (collector.hasErrors()) {
+      throw new SchemaChangeException("Failure while materializing expression. "
+          + collector.toErrorString());
+    }
+    return vector;
+  }
+
+  /**
+   * Identify the list of fields within a map which are unpivoted as columns in output
+   */
+  private void buildKeyList() {
+    List<String> lastMapKeyList = null;
+    for (VectorWrapper<?> vw : incoming) {
+      if (vw.getField().getType().getMinorType() != TypeProtos.MinorType.MAP) {
+        continue;
+      }
+
+      keyList = Lists.newArrayList();
+
+      for (ValueVector vv : vw.getValueVector()) {
+        keyList.add(vv.getField().getLastName());
+      }
+
+      if (lastMapKeyList == null) {
+        lastMapKeyList = keyList;
+      } else {
+        if (keyList.size() != lastMapKeyList.size() || !lastMapKeyList.containsAll(keyList)) {
+          throw new UnsupportedOperationException("Maps have different fields");
+        }
+      }
+    }
+  }
+
+  private void buildOutputContainer() throws SchemaChangeException {
+    dataSrcVecMap = Maps.newHashMap();
+    copySrcVecMap = Maps.newHashMap();
+    MajorType mt = null;
+
+    ErrorCollector collector = new ErrorCollectorImpl();
+    GregorianCalendar calendar = new GregorianCalendar(TimeZone.getTimeZone("UTC"));
+
+    calendar.setTimeInMillis(System.currentTimeMillis());
+    createKeyColumn("schema", ValueExpressions.getBigInt(schema++), copySrcVecMap);
+    createKeyColumn("computed", ValueExpressions.getDate(calendar), copySrcVecMap);
+
+    for (VectorWrapper<?> vw : incoming) {
+      addVectorToOutgoingContainer(vw.getField().getLastName(), vw, collector);
```

Comment:
No, we just use the input vector to obtain the MajorType and column information. We create a new output vector